File: C#/Api.Automation/tests/WooliesX.Challenge.Api.Tests/Services/OpenWeatherMapApi.cs
using Microsoft.Extensions.Configuration;
using System;
using WooliesX.Challenge.Api.Tests.Models;
using WooliesX.Challenge.Api.Tests.Utilities;

namespace WooliesX.Challenge.Api.Tests.Services
{
    public class OpenWeatherMapApi
    {
        private readonly IConfigurationRoot _configuration;
        private readonly string _url;
        private readonly string _apiKey;

        public OpenWeatherMapApi()
        {
            _configuration = TestConfiguration.GetConfiguration();
            _url = _configuration["apiUrl"];
            _apiKey = _configuration["apiKey"];
        }

        public DayOfWeek DayOfHoliday { get; set; }
        public string City { get; set; }
        public string ResponseText { get; set; }
        public OpenWeatherMapData ResponseData { get; set; }

        public void Lookup()
        {
            var uriBuilder = new UriBuilder(_url);
            uriBuilder.Query = $"q={City}&units=metric&appid={_apiKey}";

            ResponseText = RestApiUtil.Get(uriBuilder.Uri);

            if (JsonUtil.IsValidJson(ResponseText))
            {
                ResponseData = JsonUtil.Deserialize<OpenWeatherMapData>(ResponseText);
            }
        }
    }
}

File: Java/api.automation/src/test/java/wooliesx/challenge/services/OpenWeatherMapApi.java
package wooliesx.challenge.services;

import java.time.DayOfWeek;

import javax.ws.rs.client.Client;
import javax.ws.rs.client.ClientBuilder;
import javax.ws.rs.client.WebTarget;
import javax.ws.rs.core.MediaType;

import wooliesx.challenge.models.OpenWeatherMapData;

public class OpenWeatherMapApi {
	private String url; 
	private String apiKey;
	private WebTarget target;
	
	public OpenWeatherMapApi()
	{
		url = "http://api.openweathermap.org/data/2.5/forecast";
		apiKey = "38136f8cfada3e01bd71c6424c53f4be";
	}
	
	public DayOfWeek dayOfHoliday;
	
	public String city;
	
	public OpenWeatherMapData responseData;
	
	public void lookup()
	{
		Client client = ClientBuilder.newClient();
		target = client.target(url).queryParam("q", city)
                .queryParam("units", "metric")
                .queryParam("appid", apiKey);
	}
		
	public OpenWeatherMapData receive()
	{
		responseData = target.request(MediaType.APPLICATION_JSON)
        	.get(OpenWeatherMapData.class);
		
		return responseData;
	}

}
